Automattic/wp-calypso
# Components: Update JetpackHeader to support dark color scheme

When we introduced the color scheme update to Jetpack in #31141, we seem to have missed to update the partner logos. This PR takes care of that.

#### Changes proposed in this Pull Request

* Introduce a dark color scheme to the `<JetpackHeader />` component.
* Update all partner logos to support the dark color scheme.
* Add the dark color scheme specifically to the Jetpack login page (it's already always dark).
* Add the dark color scheme specifically to the Jetpack partner-branded connection pages (they're already always dark).
* Update the devdocs example to work with the dark color scheme (laser black).

#### Preview
Before:
![](https://cldup.com/x5xQUXiX56.png)

After:
![](https://cldup.com/dFUS2GtOJ0.png)

#### Testing instructions

* Checkout this branch and build Calypso.
* Go to http://calypso.localhost:3000/devdocs/design/jetpack-header and test with all color schemes.
* Verify logos look well.
* Consider the following partner IDs:
  * `51945` - Dreamhost
  * `49615` - Pressable
  * `57152` - MilesWeb
  * `58712` - Bluehost
  * `57836` - InMotion
  * `60178` - LiquidWeb
* For each partner ID, load `http://calypso.localhost:3000/log-in/jetpack?partner_id=12345` where `12345` is the partner ID, and verify it looks well.
* For each partner ID, load `http://calypso.localhost:3000/jetpack/connect?partner_id=12345` where `12345` is the partner ID, and verify it looks well.
